### PR TITLE
ENH: Mars Global Surveyor Mag data (mgs_mag)

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Adds a new instrument -- Mars Global Surveyor Magnetometer

# Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

via GitHub Actions and standard instrument tests from pysat suite.

**Test Configuration**:
* Operating system: Mac, Ubuntu, Windows
* Version number: Python 3.9, 3.10, 3.11, 3.12

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysatNASA/instruments/methods/mgs.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# TODO(#XXX): Update NRL review before next version release
```
Comment (jklenzing) [inline pysatNASA/instruments/methods/mgs.py]:
```suggestion
# TODO(#249): Update NRL review before next version release
```

File: pysatNASA/instruments/mgs_mag.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# TODO(#XXX): Update NRL review before next version release
```
Comment (jklenzing) [inline pysatNASA/instruments/mgs_mag.py]:
```suggestion
# TODO(#249): Update NRL review before next version release
```

**Review by jklenzing — COMMENTED** [review]
Linking TODO statements to newly created issue

**Comment by jklenzing** [comment]
Questions for @t-esman and @aburrell: I'm not sure whether `unix_time` or `time` is the most appropriate variable for the axis here. On Jan 1 there is an issue with non-unique time stamps, hence using Jan 9 as the test date. Not sure if that is an issue with the netcdf conversion or higher up the stream.

File: .gitignore [hunk 3]
```diff
@@ -74,3 +74,6 @@ custom_lint.sh
 
 # vscode
 .vscode
+
+# pyenv
+.python-version
```
Comment (aburrell) [inline .gitignore]:
```suggestion
.vscode

# pyenv
.python-version




```
```suggestion
.vscode

# pyenv
.python-version
```
```suggestion
.vscode

# pyenv
.python-version
```
```suggestion
.vscode

# pyenv
.python-version
```

Comment (aburrell) [reply .gitignore]: Wow.  Ok, this computer sucks.  I would like a newline added to get rid of the red symbol.

File: pysatNASA/instruments/methods/__init__.py [hunk 4]
```diff
@@ -24,3 +24,4 @@
 from pysatNASA.instruments.methods import reach  # noqa F401
 from pysatNASA.instruments.methods import ses14  # noqa F401
 from pysatNASA.instruments.methods import timed  # noqa F401
+from pysatNASA.instruments.methods import mgs  # noqa F401
```
Comment (aburrell) [inline pysatNASA/instruments/methods/__init__.py]: Same here

File: pysatNASA/instruments/methods/mgs.py (hunk 1 above)
Comment (aburrell) [reply pysatNASA/instruments/methods/mgs.py]: I think you can remove this comment for now.

File: pysatNASA/instruments/mgs_mag.py (hunk 2 above)
Comment (aburrell) [reply pysatNASA/instruments/mgs_mag.py]: Same

File: pysatNASA/instruments/mgs_mag.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+download = functools.partial(cdw.cdas_download, supported_tags=download_tags)
+
+# Set the list_remote_files routine
+list_remote_files = functools.partial(cdw.cdas_list_remote_files,
+                                      supported_tags=download_tags)
+
+# Set the load routine
+
+
+def load(fnames, tag=None, inst_id=None):
```
Comment (aburrell) [inline pysatNASA/instruments/mgs_mag.py]: Defaults should be strings.

File: pysatNASA/instruments/mgs_mag.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    Parameters
+    ----------
+    fnames : array-like
+        iterable of filename strings, full path, to data files to be loaded.
+        This input is nominally provided by pysat itself.
+    tag : str
+        tag name used to identify particular data set to be loaded.
+        This input is nominally provided by pysat itself.
+    inst_id : str
+        Satellite ID used to identify particular data set to be loaded.
```
Comment (aburrell) [inline pysatNASA/instruments/mgs_mag.py]: This should read "Instrument ID"

**Comment by aburrell** [comment]
I'm not sure either... technically 'time' is more likely to be unique, having the ability to have fractional seconds?

**Comment by t-esman** [comment]
Uhh I don't know how to use this and I did something wrong. Sorry - how do I unclose?

**Comment by jklenzing** [comment]
OK, review changes applied. Verified that 'unix_time' is unique for Jan 9 1998, 'time' is not. I'm not sure if this is a leap second or something else.

**Review by aburrell — APPROVED** [review]

**Pull request merged**